client/homebrew/brewRenderer/toolBar/toolBar.jsx

```diff
@@ -37,10 +39,67 @@ const ToolBar = ({ onZoomChange, currentPage, onPageChange, totalPages })=>{
 		setPageNum(pageNumber);
 	};
 
+
+	const calculateZoom = (mode)=>{
+		const iframe = document.getElementById('BrewRenderer');
+		const iframeWidth = iframe.getBoundingClientRect().width;
+		const iframeHeight = iframe.getBoundingClientRect().height;
+		const pages = iframe.contentWindow.document.getElementsByClassName('page');
+
+		let desiredZoom = 0;
+
+		if(mode == 'fill'){
+			// find widest page, in case pages are different widths, so that the zoom is adapted to not cut the widest page off screen.
+			let widestPage = 0;
+			[...pages].forEach((page)=>{
+				const width = page.offsetWidth;
+				if(width > widestPage)
+					widestPage = width;
+			});
+
+			desiredZoom = (iframeWidth / widestPage) * 100;
+
+		} else if(mode == 'fit'){
+			// find the page with the largest single dim (height or width) so that zoom can be adapted to fit it.
+			let maxDimRatio = 0;
+
+			[...pages].forEach((page)=>{
+				const widthRatio = iframeWidth / page.offsetWidth;
+				const heightRatio = iframeHeight / page.offsetHeight;
+				const dimensionRatio = Math.min(widthRatio, heightRatio);
+				if(dimensionRatio < maxDimRatio || maxDimRatio == 0){
+					maxDimRatio = dimensionRatio;
+				}
+			});
```

Comment:
The minimizing logic simplifies a lot if we start with `Infinity`

```suggestion
			let minDimRatio = Infinity;

			[...pages].forEach((page)=>{
				const widthRatio = iframeWidth / page.offsetWidth;
				const heightRatio = iframeHeight / page.offsetHeight;
				minDimRatio = Math.min(widthRatio, heightRatio, minDimRatio);
			});
```

or

```suggestion
			const minDimRatio = [...pages].reduce((minRatio, page) => 
				Math.min(minRatio, iframeWidth / page.offsetWidth, iframeHeight / page.offsetHeight), Infinity
			);

```

Reply:
done and done